### PR TITLE
ShowOnStartup fix

### LINKED ISSUE #38: powershell.integratedConsole.showOnStartup set to false issue
"Changing powershell.integratedConsole.showOnStartup to false disables entire functionality."

From #35 

We should verify and figure out what to do. We may need to resort to Stdio in this case if we cant hide the terminal

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Sets the buffer to hide and then hides it at start up. Also makes sure it doesn’t get revealed again via F8.

fixes #38

### CODE REVIEW

File: src/client/process.ts [hunk 1]
```diff
@@ -134,22 +137,13 @@ export class PowerShellProcess {
...
-        if (this.consoleTerminal) {
-            this.consoleTerminal.sendText(line)
-        }
-    }
 
     public async scrollToBottom() {
-        this.consoleTerminal.show(false)
-        await utils.sleep(100)
-        await vscode.workspace.nvim.command("wincmd w")
+        if (this.config.integratedConsole.showOnStartup) {
```
Comment (yatli) [inline src/client/process.ts]: what if user set showOnStartup: false but then manually show the terminal?

File: src/client/process.ts [hunk 2]
```diff
@@ -134,22 +137,13 @@ export class PowerShellProcess {
...
+            this.consoleTerminal.show(!this.config.integratedConsole.focusConsoleOnExecute)
```
Comment (yatli) [inline src/client/process.ts]: remove this line, no focus = no autoscroll :/

File: src/client/process.ts [hunk 3]
```diff
@@ -99,7 +99,10 @@ export class PowerShellProcess {
         })
 
         if (!this.config.integratedConsole.showOnStartup) {
-            this.consoleTerminal.hide()
+            // TODO: remove `as any` when coc.nvim updates past 0.0.72-471db1a4c4
+            const buf = vscode.workspace.nvim.createBuffer((this.consoleTerminal as any).bufnr);
```
Comment (yatli) [inline src/client/process.ts]: we may want to monitor the presence of the terminal buffer by registering autocommands here.

File: src/client/process.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/client/process.ts]: Good call. This was naive. Any idea if there’s a way to test if a buffer/terminal is visible?

File: src/client/process.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/client/process.ts]: :(

Comment (yatli) [reply src/client/process.ts]: But, who cares if a hidden terminal autoscrolls? :D

**Comment by TylerLeonhardt** [comment]
Ok @yatli this should be good to go now with the latest coc.nvim!

**Comment by yatli** [comment]
looking good!

**Pull request merged**